src/main/java/edu/cornell/kfs/fp/document/service/impl/CuDisbursementVoucherPayeeServiceImpl.java

```diff
@@ -146,9 +149,33 @@ public boolean isAlumni(CuDisbursementPayee payee) {
         return CuDisbursementVoucherConstants.DV_PAYEE_TYPE_ALUMNI.equals(payeeTypeCode);
     }
 
-    /**
-     * @see org.kuali.kfs.fp.document.service.DisbursementVoucherPayeeService#isVendor(org.kuali.kfs.fp.businessobject.DisbursementVoucherPayeeDetail)
-     */
+    public boolean isPayeeSignedUpForACH(DisbursementVoucherPayeeDetail disbursementVoucherPayeeDetail) {
+        boolean result = false;
+
+        if (ObjectUtils.isNotNull(disbursementVoucherPayeeDetail)) {
+            String payeeTypeCode = disbursementVoucherPayeeDetail.getDisbursementVoucherPayeeTypeCode();
+            String payeeIdNumber = disbursementVoucherPayeeDetail.getDisbVchrPayeeIdNumber();
+
+            result |= payeeACHService.isPayeeSignedUpForACH(payeeTypeCode, payeeIdNumber);
+
+            if (!result) {
+                result |= payeeACHService.isPayeeSignedUpForACH(PdpConstants.PayeeIdTypeCodes.ENTITY, getPayeeEntityId(payeeIdNumber));
+            }
+        }
+
+        return result;
+    }
     
+    protected String getPayeeEntityId(String payeeIdNumber) {
+        String entityId = StringUtils.EMPTY;
 
+        Person person = personService.getPersonByEmployeeId(payeeIdNumber);
+        if (ObjectUtils.isNotNull(person)) {
+            entityId = person.getEntityId();
+        } else {
+            entityId = payeeIdNumber;
```

Comment:
Good point, however then we will not be able to contribute this code.

I am not sure what would be the benefit of checking the disbVchrPayeeIdType here.

Reply:
According to the code in CuDisbursementVoucherDocument, the "disbVchrPayeeIdType" property will get set to "Entity ID" for students, alumni, and other users that are lacking an employee ID for some reason. This will help verify that the payeeIdNumber is indeed an entity ID.

Without some extra check for verifying the type of ID, it's possible that a vendor number or customer number could accidentally be used in a check for an entity ACH. That probably wouldn't be a risk for vendor numbers; not sure about customer numbers.

Reply:
I am not sure why it would be an issue to do a Payee ACH account search for type entity and payee number a vendor number. I can set up a payee ACH account with payee type code Entity and payee number a vendor number. There is currently no business rule in place to prevent that.

Reply:
My concern was whether there would be a significant risk of a false positive for determining Payee ACH setup. If that risk is minimal or nonexistent due to the formatting of vendor and customer numbers, then I am okay with this particular section going forward as-is.

It would be ideal to have some kind of extra verification of the ID's intended type based on the DV payee type, but I understand that the existing KualiCo code doesn't check for that when doing the employee-search failover.

Reply:
Chad,

I understand your concern.

I looked a bit more into this.

The disbVchrPayeeIdType on the extension table is just a description of the actual payee type code on the dv payee detail: disbursementVoucherPayeeTypeCode.

The code in the isPayeeSignedUpForACH method will first try to find a match based on payeeTypeCode and payeeIdNumber see line 159. If that fails it will try to find a match with payee type defaulted to ENTITY. That will only find a match for a vendor if someone in fact has set up a payee ach account with type Entity and a vendor number. In that case I think the match will be legitimate and we would want this method to return true.

Please let me know if you still have concerns and we can further discuss this and look into options.

Thank you!

Reply:
Thanks for looking into this, Daniela.

Given that the vendor numbers will likely contain dashes, and since it looks like AR customer numbers are alphanumeric and few (if any) are numeric-only, I think the risk of a false positive is minimal. I'm okay with this part moving forward as-is, and we can make changes in another user story if we run into problems with vendors or AR customers.